src/GravitationalClustering.js

```diff
@@ -0,0 +1,188 @@
+    iterations: 100,
+    seed: 42
+};
+
+/**
+ * @class GravitationalClustering
+ */
+export default class GravitationalClustering {
+
+    /**
+     * Create a new instance of a Gravitational Clustering algorithm
+     * @param {object} options
+     * @param {boolean} [options.unitMass] - the mass of each point are equal.
+     * @param {number} [options.gravitationalConstant=10^-4] - Gravitational constant.
+     * @param {number} [options.deltaGravitationalConstant=0.001] - Constant to reduce the gravitational constant.
+     * @param {number} [options.alpha=0.03] - Minimum percentage of the dataset to become a cluster.
+     * @param {number} [options.gamma=0.2] - Percentage of space used as the radius of each particle used to create the clusters.
+     * @param {string} [options.distance] - Distance function that takes two points, see [ml-distance]{@link https://github.com/mljs/distance}
+     * @param {number} [options.iterations=100] - Number of iterations of the algorithm.
+     * @param {number} [options.seed] - seed for the random generator algorithm, must be a 32-bit integer.
+     */
+    constructor(options) {
+        options = Object.assign({}, defaultOptions, options);
+        this.unitMass = options.unitMass;
+        this.gravitationalConstant = options.gravitationalConstant;
+        this.deltaGravitationalConstant = options.deltaGravitationalConstant;
+        this.alpha = options.alpha;
+        this.gamma = options.gamma;
+        this.distance = options.distance;
+        this.iterations = options.iterations;
+        this.seed = options.seed;
+
+        this.particles = [];
+        this.outliers = [];
+        this.eps = Number.MAX_VALUE;
+        this.engine = Random.engines.mt19937();
+        this.engine.seed(this.seed);
+        this.dist = Distance[this.distance];
+    }
+
+    /**
+     * Fit
+     * @param {Array} X - Points to create the clusters.
+     * @param {Array} masses - Mass of each point (Optional).
+     * @return {object}
+     */
+    train(X, masses) {
+        if (X.length === 0) {
+            throw new RangeError('The matrix length should be greater than 0.');
+        }
+        if (masses === undefined) {
+            masses = new Array(X.length);
+        }
+
+        this.particles = [];
+        this.outliers = [];
+
+        var dim = X[0].length;
+        var elements = X.length;
+        var minValues = new Array(dim);
+        var maxValues = new Array(dim);
+
+        for (var i = 0; i < dim; ++i) {
+            minValues[i] = Number.MAX_VALUE;
+            maxValues[i] = Number.MIN_VALUE;
+        }
+
+        this.uf = new UnionFind();
+        this.disjointElems = new Array(elements);
+        for (i = 0; i < elements; ++i) {
+            this.disjointElems[i] = this.uf.add(i);
+            var elem = X[i];
+            var mass = masses[i];
+            if (dim !== elem.length) {
+                throw new RangeError('The element at position ' + i + 'must have a size of '
```

Comment:
maybe with [template literals](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Template_literals) it's nicer to read